sound/soc/codecs/rt700-sdw.c

```diff
@@ -490,6 +490,24 @@ static int __maybe_unused rt700_dev_suspend(struct device *dev)
 	return 0;
 }
 
+static int __maybe_unused rt700_dev_system_suspend(struct device *dev)
+{
+	struct rt700_priv *rt700 = dev_get_drvdata(dev);
+	struct sdw_slave *slave = dev_to_sdw_dev(dev);
+
+	if (!rt700->hw_init)
+		return 0;
+
+	/*
+	 * prevent new interrupts from being handled after the
+	 * deferred work completes and before the parent disables
+	 * interrupts on the link
+	 */
+	slave->disable_interrupt_callback = true;
+
+	return rt700_dev_suspend(dev);
```

Comment:
isn't there still a race here? If the system suspend and an interrupt happen at the same time, cannot this happen on two cores like
```
core 0					core 1
system suspend
					IRQ
					flag not set
set flag
purge work queue
					schedule work
```
I don't see a way to reliable solve this apart from suspend, which would also wait for a potentially running ISR to finish

Reply:
Thanks for the comment @lyakh 
In my initial version I had an additional check for the flag in the work function.
I removed it since I thought I was overly paranoid but you convinced me that paranoia is a good thing :-)